docs/plugins/restified-endpoints/how-to.mdx

```diff
@@ -0,0 +1,253 @@
+keywords:
+  - hasura plugins
+  - plugins
+  - engine plugins
+  - custom execution
+  - custom endpoints
+  - custom handlers
+  - restified endpoints
+  - restified graphql
+seoFrontMatterUpdated: false
+---
+
+# How to Configure the RESTified Endpoints Plugin
+
+## Introduction
+
+The [RESTified GraphQL endpoints plugin](https://github.com/hasura/engine-plugin-restified-endpoint) allows you to add
+RESTified GraphQL endpoints to DDN. This can be used to add custom REST endpoints to DDN that would execute specified
+GraphQL query on the DDN GraphQL API and return the response.
+
+:::info We're using Cloudflare Wrangler
+
+In this example, we're using Cloudflare Wrangler to deploy our plugin as a Cloudflare Worker. However, you can use any
+other tool or service that hosts HTTPS services you wish. You can get started with Wrangler
+[here](https://developers.cloudflare.com/workers/wrangler/install-and-update/).
+
+:::
+
+## Step 1. Create a new plugin project
+
+Create a new Cloudflare Worker project using the `create-cloudflare` command with the
+[`restified-endpoints` plugin template](https://github.com/hasura/engine-plugin-restified-endpoint):
+
+```bash
+npx create-cloudflare@latest restified-endpoints-plugin --template https://github.com/hasura/engine-plugin-restified-endpoint
+```
+
+This will create a new project with the required files and dependencies.
+
+## Step 2. Install dependencies
+
+Navigate to the new directory and install the dependencies.
+
+```bash
+cd restified-endpoints-plugin
+npm install
+```
+
+Start the local development server.
+
+```bash
+npm start
+```
+
+## Step 3. Add the plugin configuration
+
+We'll let the engine know about the plugin and to execute it as a pre-route plugin by creating a new metadata file. In
+your `global` subgraph's metadata directory, create a new file named `restified-endpoints.hml` and add the following
+configuration.
+
+```yaml
+kind: LifecyclePluginHook
+version: v1
+definition:
+  pre: route
+  name: restified_endpoints
+  url:
+    valueFromEnv: RESTIFIED_ENDPOINTS_URL
+  config:
+    match: "/v1/api/rest/*"
+    matchMethods: ["GET", "POST"]
+    request:
+      method: GET
+      headers:
+        forward:
+          - Authorization
+          - x-hasura-role
+        additional:
+          hasura-m-auth:
+            valueFromEnv: M_AUTH_KEY
+      rawRequest:
+        path: {}
+        query: {}
+        method: {}
+        body: {}
+```
+
+:::tip URL Match
+
+The `match` field is used to match the regex to the request path, while `matchMethods` specifies the HTTP methods to
+match. In this example, we're matching all `GET` and `POST` requests to `/v1/api/rest/*`. You can modify these to match
+any path and methods (GET/POST/PUT/PATCH/DELETE) you wish.
+
+Also, note that we can not use the pre-defined DDN endpoints (like `/graphql`, `/v1/explain`, `/healthz`, `/metrics`,
+etc.) in the `match` field.
```

Comment:
Again just checking if this list is exhaustive?

Reply:
Now it is, thanks. But this is expected to change with time.